apps/src/music/utils/Loader.js

```diff
@@ -0,0 +1,101 @@
+// Utilities for retrieving various types of data for Music Lab.
+
+import {getStore} from '@cdo/apps/redux';
+import AppConfig from '../appConfig';
+import {levelsForLessonId} from '@cdo/apps/code-studio/progressRedux';
+
+export const baseUrl = 'https://curriculum.code.org/media/musiclab/';
+
+// Loads a sound library JSON file.
+export const loadLibrary = async () => {
+  if (AppConfig.getValue('local-library') === 'true') {
+    const localLibraryFilename = 'music-library';
+    const localLibrary = require(`@cdo/static/music/${localLibraryFilename}.json`);
+    return localLibrary;
+  } else {
+    const libraryParameter = AppConfig.getValue('library');
+    const libraryFilename = libraryParameter
+      ? `music-library-${libraryParameter}.json`
+      : 'music-library.json';
+    const response = await fetch(baseUrl + libraryFilename);
+    const library = await response.json();
+    return library;
+  }
+};
+
+// Loads a progression file.  This file can be used for prototyping a progression
+// before it's split into proper levels.
+const loadProgressionFile = async () => {
+  if (AppConfig.getValue('local-progression') === 'true') {
+    const defaultProgressionFilename = 'music-progression';
+    const progression = require(`@cdo/static/music/${defaultProgressionFilename}.json`);
+    return progression;
+  } else {
+    const progressionParameter = AppConfig.getValue('progression');
+    const progressionFilename = progressionParameter
+      ? `music-progression-${progressionParameter}.json`
+      : 'music-progression.json';
+    const response = await fetch(baseUrl + progressionFilename);
+    const progression = await response.json();
+    return progression;
+  }
+};
+
+// Loads level data from the dashboard, for a script level or a level.
+const loadLevelData = async levelSource => {
+  const state = getStore().getState();
+
+  let levelDataPath;
+  if (levelSource === 'levels') {
```

Comment:
> add constant for `levels` and `level`?

Added a `type`, though I'm not sure whether we prefer an `enum`?

> Also, could `levelSource` be `file` here? I see that as a `levelSource` option in `loadProgressionStepFromSource`.

No, if it's `file`, then `loadProgressionFile` is called instead of `loadLevelData`.

Reply:
👍 I think either `type` or `enum` works, although `enum` has the benefit of being able to reference a constant instead of re-typing the string. I have a slight preference for `enum` because of that, but it's a loosely held preference.

Reply:
Fair call. I'll switch to an `enum`.